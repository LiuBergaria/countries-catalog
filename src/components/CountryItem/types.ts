export interface ICountryItem {
  id: string;
  flag?: string;
  name: string;
}
