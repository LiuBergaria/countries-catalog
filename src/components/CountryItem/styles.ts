import { StyleSheet } from 'react-native';

import Colors from 'src/styles/Colors';

export const CountryItemsConstants = {
  itemHeight: 60,
};

const styles = StyleSheet.create({
  container: {
    height: CountryItemsConstants.itemHeight,
    paddingHorizontal: 24,
    flexDirection: 'row',
    alignItems: 'center',
  },
  flag: {
    width: 60,
    height: 30,
  },
  name: {
    fontSize: 16,
    marginLeft: 16,
    color: Colors.white,
    flex: 1,
  },
});

export default styles;
