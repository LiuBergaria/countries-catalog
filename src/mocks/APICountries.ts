import { IAPICountry } from 'src/services/api/countries/types';

export const mockedAPICountries = [
  {
    flags: { png: 'https://flagcdn.com/w320/se.png' },
    cca2: 'SE',
    name: { common: 'Sweden' },
  },
  {
    flags: { png: 'https://flagcdn.com/w320/al.png' },
    cca2: 'AL',
    name: { common: 'Albania' },
  },
  {
    flags: { png: 'https://flagcdn.com/w320/vu.png' },
    cca2: 'VU',
    name: { common: 'Vanuatu' },
  },
  {
    flags: { png: 'https://flagcdn.com/w320/mu.png' },
    cca2: 'MU',
    name: { common: 'Mauritius' },
  },
  {
    flags: { png: 'https://flagcdn.com/w320/gb.png' },
    cca2: 'GB',
    name: { common: 'United Kingdom' },
  },
  {
    flags: { png: 'https://flagcdn.com/w320/pt.png' },
    cca2: 'PT',
    name: { common: 'Portugal' },
  },
  {
    flags: { png: 'https://flagcdn.com/w320/ci.png' },
    cca2: 'CI',
    name: { common: 'Ivory Coast' },
  },
  {
    flags: { png: 'https://flagcdn.com/w320/ge.png' },
    cca2: 'GE',
    name: { common: 'Georgia' },
  },
  {
    flags: { png: 'https://flagcdn.com/w320/bt.png' },
    cca2: 'BT',
    name: { common: 'Bhutan' },
  },
  {
    flags: { png: 'https://flagcdn.com/w320/na.png' },
    cca2: 'NA',
    name: { common: 'Namibia' },
  },
  {
    flags: { png: 'https://flagcdn.com/w320/nl.png' },
    cca2: 'NL',
    name: { common: 'Netherlands' },
  },
  {
    flags: { png: 'https://flagcdn.com/w320/uy.png' },
    cca2: 'UY',
    name: { common: 'Uruguay' },
  },
  {
    flags: { png: 'https://flagcdn.com/w320/be.png' },
    cca2: 'BE',
    name: { common: 'Belgium' },
  },
  {
    flags: { png: 'https://flagcdn.com/w320/gg.png' },
    cca2: 'GG',
    name: { common: 'Guernsey' },
  },
  {
    flags: { png: 'https://flagcdn.com/w320/vi.png' },
    cca2: 'VI',
    name: { common: 'United States Virgin Islands' },
  },
  {
    flags: { png: 'https://flagcdn.com/w320/ug.png' },
    cca2: 'UG',
    name: { common: 'Uganda' },
  },
  {
    flags: { png: 'https://flagcdn.com/w320/li.png' },
    cca2: 'LI',
    name: { common: 'Liechtenstein' },
  },
  {
    flags: { png: 'https://flagcdn.com/w320/gw.png' },
    cca2: 'GW',
    name: { common: 'Guinea-Bissau' },
  },
  {
    flags: { png: 'https://flagcdn.com/w320/kr.png' },
    cca2: 'KR',
    name: { common: 'South Korea' },
  },
  {
    flags: { png: 'https://flagcdn.com/w320/cu.png' },
    cca2: 'CU',
    name: { common: 'Cuba' },
  },
] as IAPICountry[];
