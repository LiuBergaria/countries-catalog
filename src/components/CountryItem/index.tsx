import React from 'react';
import { View, Text } from 'react-native';

import FastImage from 'react-native-fast-image';

import styles from './styles';
import { ICountryItem } from './types';

interface IProps {
  data: ICountryItem;
}

const CountryItem = ({ data }: IProps): JSX.Element => {
  return (
    <View style={styles.container}>
      <FastImage
        style={styles.flag}
        source={{ uri: data.flag }}
        resizeMode={'contain'}
        testID="country-item-image"
      />
      <Text testID="country-item-name" style={styles.name}>
        {data.name}
      </Text>
    </View>
  );
};

export default React.memo(
  CountryItem,
  (prevProps, newProps) => prevProps.data.id === newProps.data.id,
);
