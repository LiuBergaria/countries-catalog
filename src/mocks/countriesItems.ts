import { ICountryItem } from 'src/components/CountryItem/types';

export const mockedCountryItems: ICountryItem[] = [
  { flag: 'https://flagcdn.com/w320/se.png', id: 'SE', name: 'Sweden' },
  { flag: 'https://flagcdn.com/w320/al.png', id: 'AL', name: 'Albania' },
  { flag: 'https://flagcdn.com/w320/vu.png', id: 'VU', name: 'Vanuatu' },
  { flag: 'https://flagcdn.com/w320/mu.png', id: 'MU', name: 'Mauritius' },
  { flag: 'https://flagcdn.com/w320/gb.png', id: 'GB', name: 'United Kingdom' },
  { flag: 'https://flagcdn.com/w320/pt.png', id: 'PT', name: 'Portugal' },
  { flag: 'https://flagcdn.com/w320/ci.png', id: 'CI', name: 'Ivory Coast' },
  { flag: 'https://flagcdn.com/w320/ge.png', id: 'GE', name: 'Georgia' },
  { flag: 'https://flagcdn.com/w320/bt.png', id: 'BT', name: 'Bhutan' },
  { flag: 'https://flagcdn.com/w320/na.png', id: 'NA', name: 'Namibia' },
  { flag: 'https://flagcdn.com/w320/nl.png', id: 'NL', name: 'Netherlands' },
  { flag: 'https://flagcdn.com/w320/uy.png', id: 'UY', name: 'Uruguay' },
  { flag: 'https://flagcdn.com/w320/be.png', id: 'BE', name: 'Belgium' },
  { flag: 'https://flagcdn.com/w320/gg.png', id: 'GG', name: 'Guernsey' },
  {
    flag: 'https://flagcdn.com/w320/vi.png',
    id: 'VI',
    name: 'United States Virgin Islands',
  },
  { flag: 'https://flagcdn.com/w320/ug.png', id: 'UG', name: 'Uganda' },
  { flag: 'https://flagcdn.com/w320/li.png', id: 'LI', name: 'Liechtenstein' },
  { flag: 'https://flagcdn.com/w320/gw.png', id: 'GW', name: 'Guinea-Bissau' },
  { flag: 'https://flagcdn.com/w320/kr.png', id: 'KR', name: 'South Korea' },
  { flag: 'https://flagcdn.com/w320/cu.png', id: 'CU', name: 'Cuba' },
];
