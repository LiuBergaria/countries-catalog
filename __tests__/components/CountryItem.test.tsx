import React from 'react';
import CountryItem from 'src/components/CountryItem';
import { render } from '@testing-library/react-native';
import { mockedCountryItems } from 'src/mocks/countriesItems';

test('renders CountryItem correctly', () => {
  const mappedElements = mockedCountryItems.map((country) => (
    <CountryItem key={country.id} data={country} />
  ));

  const { toJSON } = render(<>{mappedElements}</>);

  expect(toJSON()).toMatchSnapshot();
});

test('render CountryItem with image and name', () => {
  const { getByTestId } = render(<CountryItem data={mockedCountryItems[0]} />);

  const image = getByTestId('country-item-image');
  const name = getByTestId('country-item-name');

  expect(image).toBeDefined();
  expect(name).toBeDefined();
});
