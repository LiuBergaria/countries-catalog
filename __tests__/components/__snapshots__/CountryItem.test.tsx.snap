// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CountryItem correctly 1`] = `
Array [
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
        "paddingHorizontal": 24,
      }
    }
    testID="country-item"
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 30,
            "width": 60,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="contain"
        source={
          Object {
            "uri": "https://flagcdn.com/w320/se.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID="country-item-image"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "flex": 1,
          "fontSize": 16,
          "marginLeft": 16,
        }
      }
      testID="country-item-name"
    >
      Sweden
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
        "paddingHorizontal": 24,
      }
    }
    testID="country-item"
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 30,
            "width": 60,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="contain"
        source={
          Object {
            "uri": "https://flagcdn.com/w320/al.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID="country-item-image"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "flex": 1,
          "fontSize": 16,
          "marginLeft": 16,
        }
      }
      testID="country-item-name"
    >
      Albania
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
        "paddingHorizontal": 24,
      }
    }
    testID="country-item"
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 30,
            "width": 60,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="contain"
        source={
          Object {
            "uri": "https://flagcdn.com/w320/vu.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID="country-item-image"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "flex": 1,
          "fontSize": 16,
          "marginLeft": 16,
        }
      }
      testID="country-item-name"
    >
      Vanuatu
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
        "paddingHorizontal": 24,
      }
    }
    testID="country-item"
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 30,
            "width": 60,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="contain"
        source={
          Object {
            "uri": "https://flagcdn.com/w320/mu.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID="country-item-image"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "flex": 1,
          "fontSize": 16,
          "marginLeft": 16,
        }
      }
      testID="country-item-name"
    >
      Mauritius
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
        "paddingHorizontal": 24,
      }
    }
    testID="country-item"
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 30,
            "width": 60,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="contain"
        source={
          Object {
            "uri": "https://flagcdn.com/w320/gb.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID="country-item-image"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "flex": 1,
          "fontSize": 16,
          "marginLeft": 16,
        }
      }
      testID="country-item-name"
    >
      United Kingdom
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
        "paddingHorizontal": 24,
      }
    }
    testID="country-item"
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 30,
            "width": 60,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="contain"
        source={
          Object {
            "uri": "https://flagcdn.com/w320/pt.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID="country-item-image"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "flex": 1,
          "fontSize": 16,
          "marginLeft": 16,
        }
      }
      testID="country-item-name"
    >
      Portugal
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
        "paddingHorizontal": 24,
      }
    }
    testID="country-item"
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 30,
            "width": 60,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="contain"
        source={
          Object {
            "uri": "https://flagcdn.com/w320/ci.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID="country-item-image"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "flex": 1,
          "fontSize": 16,
          "marginLeft": 16,
        }
      }
      testID="country-item-name"
    >
      Ivory Coast
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
        "paddingHorizontal": 24,
      }
    }
    testID="country-item"
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 30,
            "width": 60,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="contain"
        source={
          Object {
            "uri": "https://flagcdn.com/w320/ge.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID="country-item-image"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "flex": 1,
          "fontSize": 16,
          "marginLeft": 16,
        }
      }
      testID="country-item-name"
    >
      Georgia
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
        "paddingHorizontal": 24,
      }
    }
    testID="country-item"
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 30,
            "width": 60,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="contain"
        source={
          Object {
            "uri": "https://flagcdn.com/w320/bt.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID="country-item-image"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "flex": 1,
          "fontSize": 16,
          "marginLeft": 16,
        }
      }
      testID="country-item-name"
    >
      Bhutan
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
        "paddingHorizontal": 24,
      }
    }
    testID="country-item"
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 30,
            "width": 60,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="contain"
        source={
          Object {
            "uri": "https://flagcdn.com/w320/na.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID="country-item-image"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "flex": 1,
          "fontSize": 16,
          "marginLeft": 16,
        }
      }
      testID="country-item-name"
    >
      Namibia
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
        "paddingHorizontal": 24,
      }
    }
    testID="country-item"
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 30,
            "width": 60,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="contain"
        source={
          Object {
            "uri": "https://flagcdn.com/w320/nl.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID="country-item-image"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "flex": 1,
          "fontSize": 16,
          "marginLeft": 16,
        }
      }
      testID="country-item-name"
    >
      Netherlands
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
        "paddingHorizontal": 24,
      }
    }
    testID="country-item"
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 30,
            "width": 60,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="contain"
        source={
          Object {
            "uri": "https://flagcdn.com/w320/uy.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID="country-item-image"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "flex": 1,
          "fontSize": 16,
          "marginLeft": 16,
        }
      }
      testID="country-item-name"
    >
      Uruguay
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
        "paddingHorizontal": 24,
      }
    }
    testID="country-item"
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 30,
            "width": 60,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="contain"
        source={
          Object {
            "uri": "https://flagcdn.com/w320/be.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID="country-item-image"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "flex": 1,
          "fontSize": 16,
          "marginLeft": 16,
        }
      }
      testID="country-item-name"
    >
      Belgium
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
        "paddingHorizontal": 24,
      }
    }
    testID="country-item"
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 30,
            "width": 60,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="contain"
        source={
          Object {
            "uri": "https://flagcdn.com/w320/gg.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID="country-item-image"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "flex": 1,
          "fontSize": 16,
          "marginLeft": 16,
        }
      }
      testID="country-item-name"
    >
      Guernsey
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
        "paddingHorizontal": 24,
      }
    }
    testID="country-item"
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 30,
            "width": 60,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="contain"
        source={
          Object {
            "uri": "https://flagcdn.com/w320/vi.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID="country-item-image"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "flex": 1,
          "fontSize": 16,
          "marginLeft": 16,
        }
      }
      testID="country-item-name"
    >
      United States Virgin Islands
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
        "paddingHorizontal": 24,
      }
    }
    testID="country-item"
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 30,
            "width": 60,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="contain"
        source={
          Object {
            "uri": "https://flagcdn.com/w320/ug.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID="country-item-image"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "flex": 1,
          "fontSize": 16,
          "marginLeft": 16,
        }
      }
      testID="country-item-name"
    >
      Uganda
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
        "paddingHorizontal": 24,
      }
    }
    testID="country-item"
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 30,
            "width": 60,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="contain"
        source={
          Object {
            "uri": "https://flagcdn.com/w320/li.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID="country-item-image"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "flex": 1,
          "fontSize": 16,
          "marginLeft": 16,
        }
      }
      testID="country-item-name"
    >
      Liechtenstein
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
        "paddingHorizontal": 24,
      }
    }
    testID="country-item"
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 30,
            "width": 60,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="contain"
        source={
          Object {
            "uri": "https://flagcdn.com/w320/gw.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID="country-item-image"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "flex": 1,
          "fontSize": 16,
          "marginLeft": 16,
        }
      }
      testID="country-item-name"
    >
      Guinea-Bissau
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
        "paddingHorizontal": 24,
      }
    }
    testID="country-item"
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 30,
            "width": 60,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="contain"
        source={
          Object {
            "uri": "https://flagcdn.com/w320/kr.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID="country-item-image"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "flex": 1,
          "fontSize": 16,
          "marginLeft": 16,
        }
      }
      testID="country-item-name"
    >
      South Korea
    </Text>
  </View>,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 60,
        "paddingHorizontal": 24,
      }
    }
    testID="country-item"
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "height": 30,
            "width": 60,
          },
        ]
      }
    >
      <FastImageView
        resizeMode="contain"
        source={
          Object {
            "uri": "https://flagcdn.com/w320/cu.png",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        testID="country-item-image"
      />
    </View>
    <Text
      style={
        Object {
          "color": "#ffffff",
          "flex": 1,
          "fontSize": 16,
          "marginLeft": 16,
        }
      }
      testID="country-item-name"
    >
      Cuba
    </Text>
  </View>,
]
`;
